luals-addon/factorio/factorio-plugin/util.lua

```diff
@@ -390,6 +390,7 @@ function clean_up_code_ranges()
   for i = code_ranges_count, 2, -1 do
     code_ranges[i] = nil
   end
+  code_ranges[0] = 0
```

Comment:
There's no need to reset `code_ranges[1]` to `0` because the loop above loops from `code_ranges_count` down to index `2`, so index `1` remains untouched. At the end of the day it's a code readability situation where I prefer having the loop not touch `1` to make it clear that index `1` is truly static.
I should add a comment why that first element must always exist though I just noticed.